src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -475,9 +475,15 @@ private void AppendLineNoPromptInjection(ITextBuffer buffer)
             }
 
             /// <summary>Implements <see cref="IInteractiveWindowOperations2.TypeChar"/>.</summary>
-            internal void TypeChar(char typedChar)
+            public void TypeChar(char typedChar)
             {
-                InsertText(typedChar.ToString());
+                using (var transaction = UndoHistory?.CreateTransaction(InteractiveWindowResources.TypeChar))
```

Comment:
Ah, well I guess this is an improvement then. Looking at the editor implementation of ECMD_TYPECHAR I guess it does forward to InsertText which is surprising to me. If you look at `TextTransactionMergePolicy` in the editor codebase it appears the requirements are:

1. Both transactions were initiated through EditorOperations to get the merge policy.
2. The textual descriptions are the same.
3. Edits are consecutive.

In any case, I was wrong about how I thought the editor handled this, so I guess this is better than before. I'm still curious though why we're having to do so much work here rather than forwarding to nextHandler more.